sql/creation_table.sql

```diff
@@ -1,35 +1,28 @@
-DROP TABLE IF EXISTS order_line;
-DROP TABLE IF EXISTS `order`;
-DROP TABLE IF EXISTS users;
-DROP TABLE IF EXISTS products;
-CREATE TABLE users (
+CREATE TABLE IF NOT EXISTS user (
     id int NOT NULL PRIMARY KEY AUTO_INCREMENT,
-    first_name varchar(255),
-    last_name varchar(255),
-    age int(1),
+    first_name varchar(255) NOT NULL,
+    last_name varchar(255) NOT NULL,
+    age int(2) NOT NULL,
     civil varchar(255)
-    );
+);
 
-CREATE TABLE products (
+CREATE TABLE IF NOT EXISTS product (
     id int NOT NULL PRIMARY KEY AUTO_INCREMENT,
-    price float,
-    title varchar(255)
-    );
+    price float NOT NULL,
+    title varchar(255) NOT NULL
+);
 
-
-CREATE TABLE `order` (
+CREATE TABLE IF NOT EXISTS `order` (
     id int NOT NULL PRIMARY KEY AUTO_INCREMENT,
-    user_id int,
-    date datetime,
-    FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
-    );
-
+    user_id int NOT NULL,
+    `date` datetime NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES user(id) ON DELETE CASCADE
+);
 
-CREATE TABLE order_line (
-    order_id int,
-    product_id int,
-    price float,
-    quantity int,
+CREATE TABLE IF NOT EXISTS order_line (
+    order_id int NOT NULL,
+    product_id int NOT NULL,
+    quantity int NOT NULL,
     FOREIGN KEY (order_id) REFERENCES `order`(id) ON DELETE CASCADE,
-    FOREIGN KEY (product_id) REFERENCES products(id) ON DELETE CASCADE
-    );
+    FOREIGN KEY (product_id) REFERENCES product(id) ON DELETE CASCADE
+);
```

Comment:
Quelques remarques d'ordre générale:
- toujours préférer le singulier pour le nom des tables. Même si en l'occurence une table représente une collection de choses, on essaye plutot de raisonner par rapport a la nature de ce qui est stocké (un peu comme les entités de doctrine).
- utiliser NOT NULL quand c'est possible pour éviter de te retrouver avec une base de donnée qui ressemble a un fromage plein de trous.